zerver/tests/test_email_mirror.py

```diff
@@ -287,6 +293,87 @@ def test_receive_digest_email_messages(self, mock_send_future_email, mock_enough
+
+        # Mock user activity for each user
+        realm = get_realm("zulip")
+        for realm in Realm.objects.filter(deactivated=False, show_digest_email=True):
+            for user_profile in UserProfile.objects.filter(realm=realm):
+                UserActivity.objects.create(
+                    last_visit=cutoff - datetime.timedelta(days=1),
+                    user_profile=user_profile,
+                    count=0,
+                    client=get_client('test_client'))
+
+
+
+        # Check that inactive users are enqueued
+        enqueue_emails(cutoff)
+        self.assertEqual(mock_queue_digest_recipient.call_count, 13)
+
+    @mock.patch('zerver.lib.digest.queue_digest_recipient')
+    @mock.patch('zerver.lib.digest.timezone_now')
+    def test_active_users_not_enqueued(self, mock_django_timezone, mock_queue_digest_recipient):
+        # type: (mock.MagicMock, mock.MagicMock) -> None
+
+        cutoff = timezone_now()
+        # A Tuesday
+        mock_django_timezone.return_value = datetime.datetime(year=2016, month=1, day=5)
+
+        for realm in Realm.objects.filter(deactivated=False, show_digest_email=True):
+            for user_profile in UserProfile.objects.filter(realm=realm):
+                UserActivity.objects.create(
+                    last_visit=cutoff + datetime.timedelta(days=1),
+                    user_profile=user_profile,
+                    count=0,
+                    client=get_client('test_client'))
+
+        # Check that an active user is not enqueued
+        enqueue_emails(cutoff)
+        self.assertEqual(mock_queue_digest_recipient.call_count, 0)
+
+    @mock.patch('zerver.lib.digest.queue_digest_recipient')
+    @mock.patch('zerver.lib.digest.timezone_now')
+    def test_only_enqueue_on_valid_day(self, mock_django_timezone, mock_queue_digest_recipient):
+        # type: (mock.MagicMock, mock.MagicMock) -> None
+
+        # Not a Tuesday
+        mock_django_timezone.return_value = datetime.datetime(year=2016, month=1, day=6)
+
+        # Check that digests are not sent on days other than Tuesday.
+        cutoff = timezone_now()
+        enqueue_emails(cutoff)
+        self.assertEqual(mock_queue_digest_recipient.call_count, 0)
+
+    @mock.patch('zerver.lib.digest.queue_digest_recipient')
+    @mock.patch('zerver.lib.digest.timezone_now')
+    def test_no_email_digest_for_bots(self, mock_django_timezone, mock_queue_digest_recipient):
+        # type: (mock.MagicMock, mock.MagicMock) -> None
+
+        cutoff = timezone_now()
+        # A Tuesday
+        mock_django_timezone.return_value = datetime.datetime(year=2016, month=1, day=5)
+        bot = do_create_user('some_bot@example.com', 'password', get_realm('zulip'), 'some_bot', '',
+                                     bot_type=UserProfile.DEFAULT_BOT)
+        UserActivity.objects.create(
+                    last_visit=cutoff + datetime.timedelta(days=1),
```

Comment:
this should be `-1`, right?

I suspect this means the test wasn't testing the right thing.

Reply:
Doesn't have an effect either way since bots are filtered out before users are checked for inactivity. It's just in there as a sanity check